docs/sdk/src/reference_docs/frame_pallet_coupling.rs

```diff
@@ -0,0 +1,262 @@
+//! # FRAME Pallet Coupling
+//!
+//! This reference document explains how FRAME pallets can be combined to interact together.
+//!
+//! It is suggested to re-read [`crate::polkadot_sdk::frame_runtime`], notably the information
+//! around [`frame::pallet_macros::config`]. Recall that:
+//!
+//! > The configuration trait of a pallet. Mandatory. Allows a pallet to receive types at a later
+//! > point from the runtime that wishes to contain it. It allows the pallet to be parameterized
+//! > over both types and values.
+//!
```

Comment:
I would reword this a bit to fit the story here. Also, wondering if we should use the word dependency injection which is much more common programming term.
```suggestion
//! > Configuration trait of a pallet: It allows a pallet to receive types at a later
//! > point from the runtime that wishes to contain it. It allows the pallet to be parameterized
//! > over both types and values.
//!
```